modin/pandas/groupby.py

```diff
@@ -954,14 +976,17 @@ def _default_to_pandas(self, f, *args, **kwargs):
             and len(self._by.columns) == 1
         ):
             by = self._by.columns[0] if self._drop else self._by.to_pandas().squeeze()
-        elif isinstance(self._by, type(self._query_compiler)):
+        elif self._drop and isinstance(self._by, type(self._query_compiler)):
```

Comment:
converting QC 'by' to a list of labels column only if this 'by' comes from the `self._df` (if `self._drop is True`)